### PR TITLE
Update tests to check fields set in UserAttributes and Company objects

### PR DESCRIPTION
**Pull request by brennan**
- Tests previously untested UserAttributes instances containing Company data
- Uses a matcher and assertion for tests, comparing expected with actual; this approach ensures verbose output for failed tests so identifying solutions is more streamlined
- Improves overall formatting of the class

### CODE REVIEW

File: src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java [hunk 1]
```diff
@@ -148,7 +149,20 @@ public void identifyWithCompany() {
...
+            .build();
+
+        UserAttributes expectedUserAttributes = new UserAttributes.Builder()
+            .withCompany(expectedCompany)
+            .build();
+
+        ArgumentCaptor<UserAttributes> attributesArgumentCaptor = ArgumentCaptor.forClass(UserAttributes.class);
+        verify(intercom).updateUser(attributesArgumentCaptor.capture());
+
+        assertThat(expectedUserAttributes).isEqualToComparingFieldByFieldRecursively(attributesArgumentCaptor.getValue());
```
Comment (f2prateek) [inline src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java]: Does this need `isEqualToComparingFieldByFieldRecursively`?

Comment (brennan) [reply src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java]: I'm not sure what a better alternative would be. UserAttributes doesn't include any getter methods, and we want to ensure that UserAttribute values match in each object instance. Let me know if there's something else that may work better here.

Comment (f2prateek) [reply src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java]: Is `.isEqualTo` not sufficient?

Comment (brennan) [reply src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java]: Hmmm, the test fails with `.isEqualTo` - the error indicates that new UserAttributes() =! new UserAttributes(), but definitely passes with `isEqualToComparingFieldByFieldRecursively` - let me know if I'm missing something really obvious. Thanks!

Comment (f2prateek) [reply src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java]: Ok, this is good for now then.

File: build.gradle [hunk 2]
```diff
@@ -62,5 +62,4 @@ dependencies {
   testCompile 'org.powermock:powermock-classloading-xstream:1.6.6'
 }
 
-apply plugin: 'com.f2prateek.javafmt'
```
Comment (f2prateek) [inline build.gradle:65]: Why was this removed?

File: src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java [hunk 3]
```diff
@@ -272,6 +272,7 @@ private void setUserAttributes(Traits realTraits, @Nullable Map<String, Object>
   private Company setCompany(Map<String, Object> payload) {
     Company.Builder company = new Company.Builder();
     company.withCompanyId(String.valueOf(payload.get("id")));
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java]: what if payload.id is null?

**Review by f2prateek — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (brennan) [reply build.gradle:65]: It was in the build.gradle file twice.

File: src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java (hunk 3 above)
Comment (brennan) [reply src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java]: An id is required to associate a user with a company, so if a customer doesn't include this, they're not following directions. ;)

That said, I can add an early return if the id doesn't exist to prevent a null pointer exception.

Comment (f2prateek) [reply src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java]: It won't lead to a null pointer exception; it'll just set company Id to `null`.

For the future I would point out that that `.id` is not a requirement for our spec, it's an optional field. So we should make sure to guard against that case when applicable (doesn't apply here as much though since it won't cause a NPE).

Comment (brennan) [reply src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java]: Gotcha re: null pointer.

Right, `id` isn't part of our spec, but Intercom won't create or associate a user with a company if a companyId isn't present. I think we shouldn't rely solely on *our* spec to determine whether we guard against a null.

I just added an early return if `id` isn't present for efficiency's sake.

File: src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java [hunk 4]
```diff
@@ -271,7 +271,9 @@ private void setUserAttributes(Traits realTraits, @Nullable Map<String, Object>
 
   private Company setCompany(Map<String, Object> payload) {
     Company.Builder company = new Company.Builder();
+    if (!payload.containsKey("id")) return company.build();
```
Comment (ladanazita) [inline src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java:274]: I would add a test case to see what happens when you call [`setCompany`](https://github.com/segment-integrations/analytics-android-integration-intercom/blob/40aaee3095ac40f45424019ac7a06a66b0c09d92/src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java#L255) with an empty company

File: src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java (hunk 3 above)
Comment (f2prateek) [reply src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java]: LGTM.

File: src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java (hunk 4 above)
Comment (brennan) [reply src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java:274]: Gotcha - we actually do, test class line 175!
